File: config/config-annotation-processor/src/test/java/ru/tinkoff/kora/config/annotation/processor/cases/PojoConfigRootWithComponentOf.java
package ru.tinkoff.kora.config.annotation.processor.cases;

import ru.tinkoff.kora.common.Tag;
import ru.tinkoff.kora.config.common.ConfigRoot;

@ConfigRoot(AppWithConfigWithModule.class)
public class PojoConfigRootWithComponentOf {
    private final PojoConfig pojo;
    @Tag(RecordConfig.class)
    private final RecordConfig rec;

    public PojoConfigRootWithComponentOf(PojoConfig pojo, RecordConfig rec) {
        this.pojo = pojo;
        this.rec = rec;
    }

    @Tag(PojoConfig.class)
    public PojoConfig getPojo() {
        return pojo;
    }

    public RecordConfig getRec() {
        return rec;
    }
}

File: config/config-symbol-processor/src/test/kotlin/ru/tinkoff/kora/config/symbol/processor/cases/PojoConfigRootWithComponentOf.kt
package ru.tinkoff.kora.config.symbol.processor.cases

import ru.tinkoff.kora.common.Tag
import ru.tinkoff.kora.config.common.ConfigRoot

@ConfigRoot(AppWithConfigWithModule::class)
class PojoConfigRootWithComponentOf(
    @Tag(ClassConfig::class) val pojo: ClassConfig,
    @Tag(DataClassConfig::class) val rec: DataClassConfig
)
